### PR TITLE
Add millis16 to AP_HAL - the weak form

### PR DESCRIPTION
**Pull request by peterbarker**
Oh.  Also stop Rover LED blinking too much.

### CODE REVIEW

File: libraries/AP_HAL/system.cpp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#include "system.h"
+
+uint16_t WEAK AP_HAL::millis16()
```
Comment (lucasdemarchi) [inline libraries/AP_HAL/system.cpp]: I like the use of weak symbols to provide a common implementation.

File: libraries/AP_Notify/RGBLed.cpp [hunk 2]
```diff
@@ -66,6 +66,14 @@ void RGBLed::set_rgb(uint8_t red, uint8_t green, uint8_t blue)
 // _scheduled_update - updates _red, _green, _blue according to notify flags
 void RGBLed::update_colours(void)
 {
+    // constant update rate of 10hz
+    const uint16_t now = AP_HAL::millis16();
+    const uint16_t delta = now - last_update_ms;
+    if (delta < 100) {
+        return;
+    }
+    last_update_ms = now;
```
Comment (lucasdemarchi) [inline libraries/AP_Notify/RGBLed.cpp]: can you explain why this can't be a 32bits variable and simply use millis()?

Comment (WickedShell) [reply libraries/AP_Notify/RGBLed.cpp]: I think it was an attempt to reduce memory usage (this is supposition though). There actually is a 32bit timer for the override version of the LED already that could be reused for this as a general timer and reset appropriately if overrides are active.

Comment (lucasdemarchi) [reply libraries/AP_Notify/RGBLed.cpp]: 2 bytes...? This is the same amount that could be saved by reordering the fields and avoiding the holes.

Comment (WickedShell) [reply libraries/AP_Notify/RGBLed.cpp]: I'm not saying I disagree with you, I'm just noting that's the only reason  I can come up with. I think the motivation for it was in the message interval PR where the repeated number of timestamps is a real memory savings, but here it's a bit funny I agree.

Comment (peterbarker) [reply libraries/AP_Notify/RGBLed.cpp]: @WickedShell So now you made me read more code...

We don't actually need to keep this state around, as the override function shows.

Comment (peterbarker) [reply libraries/AP_Notify/RGBLed.cpp]: @lucasdemarchi Is a `&` of a mask worth saving 2 bytes of memory?

The fact reordering the fields would save memory doesn't actually eliminate the fact using a 16-bit time does too ;-)  tridge expressed interest in using 16-bit timers more commonly, so I thought I'd try it here.

Comment (WickedShell) [reply libraries/AP_Notify/RGBLed.cpp]: Stateless is even better! :D

**Comment by peterbarker** [comment]
https://github.com/ArduPilot/ardupilot/pull/9159 fixes the problem without introducing any of the `millis16()` stuff.

**Pull request closed without merging**